src/System.Security.Permissions/tests/SecurityElementTests.cs

```diff
@@ -0,0 +1,658 @@
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+// Licensed to the .NET Foundation under one or more agreements.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections;
+using System.Globalization;
+using Xunit;
+
+namespace System.Security.Permissions.Tests
+{
+    public class SecurityElementTest
+    {
+        private static SecurityElement CreateElement()
+        {
+            SecurityElement elem = new SecurityElement("IPermission");
+            elem.AddAttribute("class", "System");
+            elem.AddAttribute("version", "1");
+
+            SecurityElement child = new SecurityElement("ConnectAccess");
+            elem.AddChild(child);
+
+            SecurityElement grandchild = new SecurityElement("ENDPOINT", "some text");
+            grandchild.AddAttribute("transport", "All");
+            grandchild.AddAttribute("host", "localhost");
+            grandchild.AddAttribute("port", "8080");
+            child.AddChild(grandchild);
+
+            SecurityElement grandchild2 = new SecurityElement("ENDPOINT");
+            grandchild2.AddAttribute("transport", "Tcp");
+            grandchild2.AddAttribute("host", "www.ximian.com");
+            grandchild2.AddAttribute("port", "All");
+            child.AddChild(grandchild2);
+
+            return elem;
+        }
+
+        [Fact]
+        public void Constructor1()
+        {
+            SecurityElement se = new SecurityElement("tag");
+            Assert.Null(se.Attributes);
+            Assert.Null(se.Children);
+            Assert.Equal("tag", se.Tag);
+            Assert.Null(se.Text);
+
+            se = new SecurityElement(string.Empty);
+            Assert.Null(se.Attributes);
+            Assert.Null(se.Children);
+            Assert.Equal(string.Empty, se.Tag);
+            Assert.Null(se.Text);
+        }
+
+        [Fact]
+        public void Constructor1_Tag_Invalid()
+        {
+            try
+            {
+                new SecurityElement("Na<me");
+                Assert.False(true);
+            }
+            catch (ArgumentException ex)
+            {
+                // Invalid element tag Nam<e
+                Assert.Equal(typeof(ArgumentException), ex.GetType());
+                Assert.Null(ex.InnerException);
+                Assert.NotNull(ex.Message);
+                Assert.True(ex.Message.IndexOf("Na<me") != -1);
```

Comment:
This, and other, related tests, should use one const string rather than two literals... so the test doesn't get out of sync with itself